Add layout design for current weather

minaahilimtiaz opened the pull request:
Add recyclerview in home fragment

Add cardview and constraintlayout for design

Add adapter and viewholder classes

minaahilimtiaz committed:
Add layout design for current weather

Add recyclerview in home fragment

Add cardview and constraintlayout for design

Add adapter and viewholder classes

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt:
```diff
@@ -0,0 +1,69 @@
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Remove extra code

minaahilimtiaz replied:
Removed the auto-generated code from HomeScreenFragment Class

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt:
```diff
@@ -0,0 +1,69 @@
+ * Use the [HomeScreenFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class HomeScreenFragment : Fragment() {
```
Create base class for fragment

minaahilimtiaz replied:
Added base class for fragment and activity

noumanahmad001 commented on app/src/main/res/layout/current_weather_layout.xml:
```diff
@@ -0,0 +1,135 @@
+                android:orientation="horizontal"
+                app:layout_constraintGuide_begin="2dp" />
+
+            <TextView
```
Too much properties there, try to move these to styles

minaahilimtiaz replied:
Added style for cardview and textview

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt:
```diff
@@ -0,0 +1,69 @@
+         * @return A new instance of fragment HomeScreenFragment.
+         */
+        // TODO: Rename and change types and number of parameters
+        @JvmStatic
```
If you are not using any param then please remove the ARG_PARAMs

minaahilimtiaz replied:
Yes not using it. Removed it.

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt:
```diff
@@ -0,0 +1,69 @@
+                "Abottabad, Pakistan", "Friday, August 18, 2020"))
+
+        val adapter= HomeScreenAdapter()
+        binding.homeScreenRcv.adapter=adapter
```
Rename this homeScreenRcv to more meaningful name . 
Add space before and after '='

minaahilimtiaz replied:
Renamed it to ScreenRecyclerview and added space before and '='

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt:
```diff
@@ -0,0 +1,69 @@
+
+        val adapter= HomeScreenAdapter()
+        binding.homeScreenRcv.adapter=adapter
+        adapter.data= list
```
there should be space before =

minaahilimtiaz replied:
Added space before and after '='

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt:
```diff
@@ -0,0 +1,53 @@
+
+class HomeScreenAdapter:RecyclerView.Adapter<HomeScreenAdapter.ViewHolder>() {
+
+    var data = listOf<CurrentWeatherModel>()
```
variable data should be renamed.

minaahilimtiaz replied:
renamed it to **currentWeatherList**

noumanahmad001 commented on app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt:
```diff
@@ -0,0 +1,53 @@
+        holder.bindingViewToData(item)
+    }
+
+    class ViewHolder (itemView: View): RecyclerView.ViewHolder(itemView) {
```
ViewHolder class should be renamed.

minaahilimtiaz replied:
Renamed it to **CurrentWeatherViewHolder**

AbdulMateenChughtai commented on app/src/main/java/com/example/android/weatherapp/models/CurrentWeatherModel.kt:
```diff
@@ -0,0 +1,10 @@
+package com.example.android.weatherapp.models
+
+import android.service.controls.templates.TemperatureControlTemplate
```
What is this import supposed to do?

minaahilimtiaz replied:
It is an unused import added while adding **temperature** variable. I have removed it.

AbdulMateenChughtai commented on app/src/main/res/layout/current_weather_layout.xml:
```diff
@@ -0,0 +1,135 @@
+                app:layout_constraintVertical_bias="0.5"
+                app:srcCompat="@drawable/sunny_image" />
+
+            <androidx.constraintlayout.widget.Barrier
```
For now, it's okay, but you can achieve almost this layout by using just one guideline. Always try to keep xml files as lean as possible.

minaahilimtiaz replied:
Should i leave it for now or change it?

AbdulMateenChughtai replied:
Keep it as is for now.

AbdulMateenChughtai commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,12 @@
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+    <fragment
```
Create fragments dynamically/programmatically.

minaahilimtiaz committed:
Add style for cardview and textview

Remove unused import from CurrentWeatherModel class

Add base class for fragment and activity

Remove unused code from HomeScreenFragment

Rename list variable to currentWeatherList in HomeScreenFragment

Rename viewholder class to CurrentWeatherViewHolder
